src/main/java/net/fabricmc/loader/util/version/SemanticVersionImpl.java

```diff
@@ -215,4 +216,66 @@ public boolean equalsComponentsExactly(SemanticVersionImpl other) {
 	boolean isPrerelease() {
 		return prerelease != null;
 	}
+
+	@Override
+	public int compareTo(SemanticVersion o) {
+		for (int i = 0; i < Math.max(getVersionComponentCount(), o.getVersionComponentCount()); i++) {
+			int first = getVersionComponent(i);
+			int second = o.getVersionComponent(i);
+			if (first == COMPONENT_WILDCARD || second == COMPONENT_WILDCARD) {
+				continue;
+			}
+
+			int compare = Integer.compare(first, second);
+			if (compare != 0) {
+				return compare;
+			}
+		}
+
+		Optional<String> prereleaseA = getPrereleaseKey();
+		Optional<String> prereleaseB = o.getPrereleaseKey();
+
+		if (prereleaseA.isPresent() || prereleaseB.isPresent()) {
+			if (prereleaseA.isPresent() && prereleaseB.isPresent()) {
+				StringTokenizer prereleaseATokenizer = new StringTokenizer(prereleaseA.get(), ".");
+				StringTokenizer prereleaseBTokenizer = new StringTokenizer(prereleaseB.get(), ".");
+
+				while (prereleaseATokenizer.hasMoreElements()) {
+					if (prereleaseBTokenizer.hasMoreElements()) {
+						String partA = prereleaseATokenizer.nextToken();
+						String partB = prereleaseBTokenizer.nextToken();
+
+						int compare;
+						try {
+							int numA = Integer.parseUnsignedInt(partA);
+							try {
+								int numB = Integer.parseUnsignedInt(partB);
+								compare = Integer.compareUnsigned(numA, numB);
+							} catch (NumberFormatException e) {
+								compare = -1;
```

Comment:
IMO this shouldn't rely on parse exceptions, but check explicitly against the spec with a regex like `0|[1-9]\d*`

Reply:
How would one handle the exceptions then?

Reply:
Is the updated version better?